src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -376,17 +372,69 @@ internal void DoGetExtensionMethods(ArrayBuilder<MethodSymbol> methods, string n
         }
 
 #nullable enable
+        internal static bool IsInvalidExtensionReceiverParameter(ParameterSymbol thisParam)
```

Comment:
Can this be private? Also consider inverting the logic and simplifying this to "is _valid_ extension receiver parameter".